course_access_groups/acl_backends.py

```diff
@@ -1,19 +1,68 @@
 """
 Access Control backends to implement the Course Access Groups.
 """
-import six
 
+from django.conf import settings
+from organizations.api import get_course_organizations
+from organizations.models import UserOrganizationMapping
+from course_access_groups.models import (
+    CourseAccessGroup,
+    GroupCourse,
+    Membership,
+)
 
-def dummy_backend(user, resource, default_has_access, options):  # pylint: disable=unused-argument
+
+def is_organization_staff(user, course):
```

Comment:
Does it make more sense to have "site staff" than "organization staff"? Will we support multiple organizations per site?

Reply:
Good question. It's an open question. Nevertheless I'm leaning towards org-based at the moment. Please consider checking this issue:

 - [Open Question: Sites vs. Organizations: Which way we should split our courses? #8
](https://github.com/appsembler/course-access-groups/issues/8).

I'll volunteer both of us for it.

Reply:
This is a huge piece of tech debt. In the design stages of Tahoe, the idea was a relation between Organizations, sites and user like this:
An Organization can "contains" multiple Sites, and ultimately Sites can "contains" multiple users. This was aiming to have "agencies" administrating more than one site, for reselling purposes. The truth is it was never implemented, and all the users across the edx platforma and edx organizations are mapped to organizations and not to sites, it also all the checks for suer belonging are made against organization and not site. This is terrible from the design point, but is tech debt, not by design. 

So @OmarIthawi I'd say, stick to org and ignore site.

Reply:
That is interesting. I've interpreted the other way around, that a site could contain multiple orgs

Anyway, yeah, looks like more than a couple of hours of design work, so I'll go with sticking with org. However, i think we should try to make sure we don't future proof an organizations update against user-site mapping in the future,

Reply:
We gotta document those stuff and make sure everyone is one the same page. I myself just went with the flow and did what works. Thanks John again for asking such critical questions.